packages/liveblocks-core/src/room.ts

```diff
@@ -1016,7 +1016,9 @@ function makeStateMachine<
       if (context.socket === null) {
         throw new Error("Can't send message if socket is null");
       }
-      context.socket.send(JSON.stringify(messageOrMessages));
+      if (context.socket.readyState === context.socket.OPEN) {
```

Comment:
What we're _really_ checking here is how far along its lifecycle this WebSocket is. In a way, we're checking

```suggestion
      if (context.socket.readyState >= context.socket.OPEN) {
```

here. Since the state diagram for WebSockets looks like this:

```mermaid
stateDiagram-v2
    direction LR
    [*] --> CONNECTING
    CONNECTING --> OPEN
    OPEN --> CLOSING
    CLOSING --> CLOSED
    CLOSED --> [*]
```

There is no way to "go back" a phase.